capstone-frontend/src/components/SearchUserModal.jsx

```diff
@@ -69,9 +72,83 @@ export class SearchUserModal extends Component {
     })
   }
 
+  componentDidMount() {
+    this.fetchCollaborators();
+  }
+
+  fetchCollaborators = async () => {
+
+    if (!(this.props.type === 'booklists' && this.props.bookList)) {
+      return;
+    }
+
+    else if (this.props.bookList.collaboratorsIDs.length === 0) {
+      return;
+    }
+
+    let collaborators = [];
+
+    await Promise.all(this.props.bookList.collaboratorsIDs.map(async (collaboratorId) => {
+
+      const collaborator = await fetch(`/api/user?id=${collaboratorId}`).then(resp => resp.json());
+
+      delete collaborator.tokenObj;
+      collaborators.push(collaborator);
+
+    }));
+
+    // Checks for owner
+    collaborators = collaborators.filter((collaborator) => collaborator.id !== this.props.bookList.id);
+
+    this.setState({ addedUsersTracker: collaborators, addedUsers: collaborators });
+
+  }
+
+  arrayContainsJSONId = (array, json) => {
+
+    for (const item of array) {
+      if (item.id === json.id) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  addUserToBookListCollaborators = (bookList, user) => {
+
+    const bookListUpdateJson = {
+      id: bookList.id,
+      add_collaboratorsIDs: user.id,
+    }
+
+    // Add Collaboartor to Booklist in Firebase
```

Comment:
nit: `Collaborator` instead of `Collaboartor`

Reply:
Done.